datacite-findable-doi-handler/src/main/java/no/unit/nva/datacite/handlers/PublicationApiClient.java

```diff
@@ -0,0 +1,59 @@
+package no.unit.nva.datacite.handlers;
+
+import static nva.commons.core.attempt.Try.attempt;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.attempt.Failure;
+
+public class PublicationApiClient {
```

Comment:
This class takes an URI and fetches dataCiteMetaData from it according to some contract (content type). I would choose a name that reflects its purpose rather than naming it according to the API we know is the one providing the data. E.g. DataCiteMetadataResolver or something.